src/backend/allocator/gbm.rs

```diff
@@ -0,0 +1,140 @@
+//! Module for Buffers created using [libgbm](reexports::gbm)
+
+use super::{
+    dmabuf::{AsDmabuf, Dmabuf},
+    Allocator, Buffer, Format, Fourcc, Modifier,
+};
+use gbm::{BufferObject as GbmBuffer, BufferObjectFlags, Device as GbmDevice};
+use std::os::unix::io::AsRawFd;
+
+impl<A: AsRawFd + 'static, T> Allocator<GbmBuffer<T>> for GbmDevice<A> {
+    type Error = std::io::Error;
+
+    fn create_buffer(&mut self, width: u32, height: u32, format: Format) -> std::io::Result<GbmBuffer<T>> {
+        if format.modifier == Modifier::Invalid || format.modifier == Modifier::Linear {
+            let mut usage = BufferObjectFlags::SCANOUT | BufferObjectFlags::RENDERING;
+            if format.modifier == Modifier::Linear {
+                usage |= BufferObjectFlags::LINEAR;
+            }
+            self.create_buffer_object(width, height, format.code, usage)
+        } else {
+            self.create_buffer_object_with_modifiers(
+                width,
+                height,
+                format.code,
+                Some(format.modifier).into_iter(),
+            )
+        }
+    }
+}
+
+impl<T> Buffer for GbmBuffer<T> {
+    fn width(&self) -> u32 {
+        self.width().unwrap_or(0)
+    }
+
+    fn height(&self) -> u32 {
+        self.height().unwrap_or(0)
+    }
+
+    fn format(&self) -> Format {
+        Format {
+            code: self.format().unwrap_or(Fourcc::Argb8888), // we got to return something, but this should never happen anyway
```

Comment:
I have a feeling that if this even happens, it'll be a nightmare to debug...

Reply:
Not really. If this happens the file descriptor of the gbm device has become invalid. So we cannot actually submit any rendering, that could be based on this format and the chain will fail sometime later anyway.